src/posts/2020-09-19-how-to-write-accessible-forms.md

```diff
@@ -0,0 +1,97 @@
+---
+layout: post
+title: Create accessible form(s)
+description: Practial tips on how to create accessible forms.
+category: How-to
+author: Hamsa Harcourt
+date: 2020-09-16
+further_reading:
+  - title: HTML label element reference
+    url: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label
+    source: MDN web docs
+    year: 2020
+  - title: Don’t Use The Placeholder Attribute
+    url: https://www.smashingmagazine.com/2018/06/placeholder-attribute/
+    source: Smashing Magazine
+    year: 2018
+  - title: "HTML5 Accessibility Chops: the placeholder attribute"
+    url: https://developer.paciellogroup.com/blog/2011/02/html5-accessibility-chops-the-placeholder-attribute/
+    source: The Paciello Group
+    year: 2011
+  - title: Using @placeholder on Input
+    url: https://www.w3.org/WAI/GL/wiki/Using_@placeholder_on_input
+    source: W3C
+    year: 2012
+eleventyNavigation:
+  key: {{ category }}
+  title: {{ title }}
+tags:
+  - howto
+---
+
+With the world becoming data reliant, it's no surprise how data is the new gold. With the right data, you can improve peoples' lives, find solutions to problems and so on. There are many ways to capture users' data but on the web, forms are your go-to solution.
+
+Here are some  practical tips on how to write accessible forms.
+
+## Always label your forms
+Labels are a great way to get started with writing accessible forms. They are vital for visually impaired people as they help [screen readers](https://en.wikipedia.org/wiki/Screen_reader) determine what label text go with what input field.
+
+There are two ways to associate a `<label>` with an `<input>`.
+- Set the identifier(ID) in the `<input>` to correspond with the `for` attribute of the `<label>`.
+```html
+<form>
+    <label for="firstName">First Name</label>
+    <input type="text" id="firstName" name="firtName"/>
+</form>
+```
+
+- Nest `<input>` directly inside `<label>`.
+```html
+<form>
+    <label>
+        First Name
+        <input type="text" name="firstName">
+    </label>
+</form>
+```
+
+
+## Highlight input element on focus
+Form cusors can be difficult to track. A simple solution is to highlight `<input>` on focus. This way, people do not have to guess where they are on the form. 
+
+![A form with an input field highlighted in blue.](https://res.cloudinary.com/dgn6edv1k/image/upload/v1600204220/a11y_anz7bn.png)
+
+```css
+input:focus{
+    outline: 2px solid royalblue
+    box-shadow: 1px 1px 8px 1px royalblue
+}
+```
+
+## Break long forms into smaller sections
+Filling long forms can be stressful. This is relatably true for almost everyone but perculiar to people  with [AD/HD](https://www.cdc.gov/ncbddd/adhd/facts.html).  They can lead to an increase in [Bounce rate](https://en.wikipedia.org/wiki/Bounce_rate) which is not ideal. A simple solution is to break them into smaller sections. These sections should include a progress bar.

```

Comment:
```suggestion
Filling long forms can be stressful. This is largely true for almost everyone but peculiarly for people with [AD/HD](https://www.cdc.gov/ncbddd/adhd/facts.html).  Such forms can lead to an increase in [Bounce rate](https://en.wikipedia.org/wiki/Bounce_rate) (i.e., people abandoning the form) which isn't what anyone wants. A simple solution is to break them into smaller sections. These sections could include a progress bar to indicate how far along a user is in the form process.

```

Reply:
made some wording suggestions here.  this could still stand to have more factual information behind it.  e.g., _why_ is it better to split a form into various different parts.  it's still a long form, just now spread across multiple screens.  so _how_ is that helpful?   info like this would be useful.

Reply:
It sounds like you're talking about the [progressive disclosure pattern](https://www.nngroup.com/articles/progressive-disclosure/). If we're going to be recommending this, it would be helpful to include a little more detail about when and when not to utilize this pattern.